apps/aecore/lib/aecore/oracle/oracle.ex

```diff
@@ -218,18 +224,17 @@ defmodule Aecore.Oracle.Oracle do
                                                                         acc ->
       if calculate_absolute_ttl(tx.ttl, height_included) <= block_height do
         acc
+        |> Chainstate.destruct()
         |> pop_in([:oracles, :registered_oracles, address])
```

Comment:
Acces.key(oracles) can be used instead of destructing the chainstate, this will remove the to_struct line as well